src/Cli/dotnet/commands/dotnet-tool/install/ParseResultExtension.cs

```diff
@@ -36,6 +30,13 @@ public static VersionRange GetVersionRange(this ParseResult parseResult)
             }
 
             VersionRange versionRange = null;
+
+            // accept 'bare' versions and interpret 'bare' versions as NuGet exact versions
+            if (!string.IsNullOrEmpty(packageVersion) && NuGetVersion.TryParse(packageVersion, out SemanticVersion version2))
+            {
+                return new VersionRange(minVersion: new NuGetVersion(packageVersion), includeMinVersion: true, maxVersion: new NuGetVersion(packageVersion), includeMaxVersion: true, originalString: "[" + packageVersion + "]");
```

Comment:
You only need to parse the version string once, not 3 times

```suggestion
            if (!string.IsNullOrEmpty(packageVersion) && NuGetVersion.TryParse(packageVersion, out NuGetVersion version2))
            {
                return new VersionRange(minVersion: version2, includeMinVersion: true, maxVersion: version2, includeMaxVersion: true, originalString: "[" + packageVersion + "]");
```